crates/bevy_utils/src/anystack.rs

```diff
@@ -0,0 +1,218 @@
+struct AnyStackMeta {
+    offset: usize,
+    func: unsafe fn(value: *mut u8, user_data: *mut u8),
```

Comment:
I think we could just be generic over the type of the `user_data`

Reply:
Yep, that was my thought when I said 
> code around the user_data: *mut u8.

The main reason I didn't limit it to some generic type is that I didn't want to force the type to be either a copy, ref, or mut ref. 
Though I'll probably just make it generic over some type to reduce the unsafe :)